src/nvim/terminal.c

```diff
@@ -907,6 +928,16 @@ static void on_refresh(Event event)
       }
       continue;
     }
+    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+      if (wp->w_buffer == term->buf) {
+        if ( wp->w_botline > wp->w_buffer->b_ml.ml_line_count ) {
+          // if last line is visible, follow the terminal
+          wp->following_terminal = true;
+        } else {
+          wp->following_terminal = false;
+        }
```

Comment:
Typically, terminals only scroll when the cursor is at the last line, Also, not sure I want unfocused windows being scrolled. I might split a terminal buffer for the specific purpose of keeping the cursor at its current position as an alternative to running `$ cmd-with-log-output | less`.